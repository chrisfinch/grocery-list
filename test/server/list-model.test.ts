import ListModel from './../../src/server/models/lists.model';

describe('List Model', () => {
    it('Runs getLists correctly', () => {
        const result = ListModel.getLists();

        expect(result.length).toBe(2);
        expect(result).toMatchSnapshot();
    });

    it('Runs getList correctly', () => {
        const result = ListModel.getList('1');

        expect(result.title).toBe('Demo list');
        expect(result).toMatchSnapshot();
    });

    it('Runs addList correctly', async () => {
        const newTitle = 'TestList';

        const result = await ListModel.addList({
            title: newTitle,
            created: 123
        });

        expect(result.id.length).toBeGreaterThan(0);
        expect(result.title).toBe(newTitle);
    });

    it('Runs deleteList correctly', () => {
        const result = ListModel.deleteList('2');

        expect(result).toBe(true);
    });

    it('Runs updateList correctly', () => {
        const newTitle = 'TestList';

        const result = ListModel.updateList({
            id: '1',
            title: newTitle
        });

        expect(result).toBe(true);
    });
});
