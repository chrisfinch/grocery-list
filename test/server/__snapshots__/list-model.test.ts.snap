// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Model Runs getList correctly 1`] = `
Object {
  "created": 1564695374960,
  "id": "1",
  "items": Array [
    Object {
      "name": "beans",
      "purchased": false,
    },
    Object {
      "name": "bread",
      "purchased": false,
    },
    Object {
      "name": "broccoli",
      "purchased": false,
    },
    Object {
      "name": "cereal",
      "purchased": false,
    },
    Object {
      "name": "squash",
      "purchased": false,
    },
  ],
  "title": "Demo list",
}
`;

exports[`List Model Runs getLists correctly 1`] = `
Array [
  Object {
    "created": 1564695374960,
    "id": "1",
    "size": 5,
    "title": "Demo list",
  },
  Object {
    "created": 1564695541598,
    "id": "2",
    "size": 1,
    "title": "Other list",
  },
]
`;
